man/Makefile

```diff
@@ -0,0 +1,41 @@
+prefix = /usr/local
+mandir = $(prefix)/man
+INSTALL = install
+INSTALL_DATA = ${INSTALL} -m 644
+
+ALL_PAGES := $(wildcard *.*.md)
+
+# Determine which manual sections we are generating pages for
+# by isolating the last part of the filename before the extension
+# and eliminating duplicates.
+man_section = $(lastword $(subst ., ,$(1)))
+sections := $(sort $(foreach page,$(ALL_PAGES:.md=),$(call man_section,$(page))))
+
+# Dynamically generate pattern rules for each manual section
+# so make knows how to build a target like man8/dockerd.8.
+define MANPAGE_template
+man$(1)/%.$(1): %.$(1).md .build/go-md2man | man$(1)
```

Comment:
How would you envision using this rule in places where there's already an appropriate `go-md2man` installed, such as in packaging (esp. where downloading/installing during build is inappropriate)?  My cleanest answer given this code as-is would be to symlink the distribution-provided `go-md2man` binary into `.build/go-md2man` so this is satisfied or patching this `Makefile` directly to remove this dependency, but neither is exactly super great. :thinking:

I guess the vendored dependencies you've included here help for my own personal builds, but _won't_ help with getting this into the Debian builds, for example (since they'll have to strip `vendor/`), so I think it's still worth thinking about whether there's a better way we can enable using a pre-existing `go-md2man` here.

Reply:
Maybe a `GO_MD2MAN` variable or something that when set is used as the `go-md2man` implementation instead of building one?  Then we'd do something like `make GO_MD2MAN=go-md2man all` and the Complicated Bits would live in the `Makefile` instead with an explicit signal-of-intent?

Reply:
Done!